src/main/java/de/hysky/skyblocker/skyblock/tabhud/widget/hud/HudCommsWidget.java

```diff
@@ -54,17 +56,97 @@ public void updateContent() {
             }
 
             Component comp;
+            var max = getCommissionMax(comm.commission());
             if (isFancy) {
-                comp = new ProgressComponent(Ico.BOOK, c, p, Colors.pcntToCol(p));
+                if (SkyblockerConfigManager.get().mining.dwarvenHud.showNumbers && max != null) {
+                    comp = new ProgressComponent(Ico.BOOK, c, p, max, Colors.pcntToCol(p));
+                } else {
+                    comp = new ProgressComponent(Ico.BOOK, c, p, Colors.pcntToCol(p));
+                }
             } else {
-                comp = new PlainTextComponent(
-                        Text.literal(comm.commission() + ": ").append(
-                                Text.literal(comm.progression()).withColor(Colors.pcntToCol(p))
-                        )
-                );
+                if (SkyblockerConfigManager.get().mining.dwarvenHud.showNumbers && max != null) {
+                    comp = new PlainTextComponent(
+                            Text.literal(comm.commission() + ": ").append(
+                                    Text.literal(p == 100f ? "DONE" : Math.round(max * (p / 100)) + "/" + max).withColor(Colors.pcntToCol(p))
+                            )
+                    );
+                } else {
+                    comp = new PlainTextComponent(
+                            Text.literal(comm.commission() + ": ").append(
+                                    Text.literal(comm.progression()).withColor(Colors.pcntToCol(p))
+                            )
+                    );
+                }
             }
             this.addComponent(comp);
         }
     }
 
+    /**
+     * Gets the actions needed to complete a commission
+     *
+     * @param commission the string name of the commission
+     * @return the actions needed to complete the commission
+     */
+    private static Integer getCommissionMax(String commission) {
```

Comment:
Please don't box the int. The default value for ints should be 0. Also, a map here would work better and you can use `Map#getOrDefault(commission, 0)`.

Reply:
or OptionalInt if you wanna be fancy ooOoOooo

Reply:
The reason I did not use a map is because some commissions have location specific logic to determine the amount (there are two coms called goblin slayer for example), and it is easier to centralize that logic inside a switch statement, whereas with a map you would need to have that location specific logic at every point the map is acessed

Reply:
My bad, I failed to notice that.